api/controllers/game/resolve.js

```diff
@@ -166,23 +164,22 @@ module.exports = function (req, res) {
 
                 // If more cards are left in deck, replace second card with card from deck
                 if (game.deck.length >= 1) {
-                  let newSecondCard = _.sample(game.deck).id;
-                  // Ensure new second card is distinct from cards drawn and new top card
-                  while (cardsToRemoveFromDeck.includes(newSecondCard)) {
-                    newSecondCard = _.sample(game.deck).id;
-                  }
+                  const newSecondCardIndex = _.random(game.deck.length - 1);
+                  const newSecondCard = game.deck.splice(newSecondCardIndex, 1)[0].id;
                   gameUpdates.secondCard = newSecondCard;
+                  cardsToRemoveFromDeck.push(newSecondCard);
                   gameUpdates.log = [
                     ...game.log,
                     `The ${game.oneOff.name} one-off resolves; ${player.username} draws one card to reach the hand limit (8).`,
                   ];
-                  // Player draws last card in deck, to reach hand limit (only draws 1)
-                } else {
-                  gameUpdates.log = [
-                    ...game.log,
-                    `The ${game.oneOff.name} one-off resolves; ${player.username} draws one card (last in deck) to reach the hand limit (8).`,
-                  ];
+
+                // Player draws last card in deck, to reach hand limit (only draws 1)
                 }
+              } else {
+                gameUpdates.log = [
+                  ...game.log,
+                  `The ${game.oneOff.name} one-off resolves; ${player.username} draws one card (last in deck) to reach the hand limit (8).`,
+                ];
```

Comment:
This was previously in the wrong place and would never fire; a new test case made that clear.

Reply:
I considered adding an assertion about the log to the test, but skipped it. We're not testing the log in other places and it's somewhat iffy how important this distinction is here. If anything needs log testing it'd be the resolution of 3's and 4's where the log is the only way to see which cards were selected